xgboost/notebooks/Online Prediction with XGBoost.ipynb

```diff
@@ -0,0 +1,634 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Online Prediction with XGBoost on Cloud Machine Learning Engine\n",
+    "This notebook uses the [Census Income Data Set](https://archive.ics.uci.edu/ml/datasets/Census+Income) to create a simple XGBoost model, upload the model to Cloud Machine Learning Engine (ML Engine), and query it for predictions. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# How to bring your model to ML Engine\n",
+    "Getting your model ready for predictions can be done in 5 steps:\n",
+    "1. Save your model to a file\n",
+    "2. Upload the saved model to [Google Cloud Storage](https://cloud.google.com/storage)\n",
+    "3. Create a model resource on ML Engine\n",
+    "4. Create a model version (linking your XGBoost model)\n",
+    "5. Make an online prediction"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Prerequisites\n",
+    "Before we begin, let’s cover some of the different tools you’ll use to get online prediction up and running on ML Engine. \n",
+    "\n",
+    "[Google Cloud Platform](https://cloud.google.com/) (GCP) lets you build and host applications and websites, store data, and analyze data on Google's scalable infrastructure.\n",
+    "\n",
+    "[Cloud ML Engine](https://cloud.google.com/ml-engine/) is a managed service that enables you to easily build machine learning models that work on any type of data, of any size.\n",
+    "\n",
+    "[Google Cloud Storage](https://cloud.google.com/storage/) (GCS) is a unified object storage for developers and enterprises, from live data serving to data analytics/ML to data archiving.\n",
+    "\n",
+    "[Cloud SDK](https://cloud.google.com/sdk/) is a set of tools for Cloud Platform. It contains gcloud, gsutil, and bq, which you can use to access Google Compute Engine, Google Cloud Storage, Google BigQuery, and other products and services from the command-line.\n"
```

Comment:
We should make it more explicit that we assume the Cloud SDK to be installed in the environment that Jupyter is running in - for example, if a user decides to run Jupyter in a Docker container, it is unlikely to have the `gcloud` command available to it and all the `! gcloud ...` cells will fail.

Reply:
We can link them to the installation page, would that suffice?
https://cloud.google.com/sdk/downloads

Reply:
No, linking isn't enough. I think we need to have a (very) short explanation that `gcloud` should be available on the same machine that they run their Jupyter notebook server on.

Reply:
Note: If you get an error below, make sure the Cloud SDK is installed in the kernel's environment.

^how's that

Reply:
I prefer something like replacing:

> [Cloud SDK](https://cloud.google.com/sdk/) is a set of tools for Cloud Platform. It contains gcloud, gsutil, and bq, which you can use to access Google Compute Engine, Google Cloud Storage, Google BigQuery, and other products and services from the command-line.

with

> [Cloud SDK](https://cloud.google.com/sdk/) is a command line tool which allows you to interact with Google Cloud products. In order to run this notebook, make sure that Cloud SDK is [installed](https://cloud.google.com/sdk/downloads) in the same environment as your Jupyter kernel.